### ISSUE TITLE
Add ability to sort by column in Collections Info mode

### ISSUE THREAD

**Issue by EdwardMoyse** [HSF/phoenix#291]
It would be helpful to e.g. click on the pT column and sort by this, to find the most energetic tracks

![image](https://user-images.githubusercontent.com/6764617/116141003-686c3200-a6d8-11eb-920b-4a555258cd0b.png)

**Comment by Ayushsunny** [comment]
Hello Sir @EdwardMoyse, can you guide me on this? I would like to work on this issue

**Comment by EdwardMoyse** [comment]
Hi @Ayushsunny - sure! The code to draw this dialog is here:
https://github.com/HSF/phoenix/tree/master/packages/phoenix-ng/projects/phoenix-ui-components/src/components/ui-menu/collections-info/collections-info-overlay

So the idea would be that for each column, we add an arrow to allow sorting the list by that column. The default should be the number (i.e. the current behaviour).

**Comment by Ayushsunny** [comment]
> Hi @Ayushsunny - sure! The code to draw this dialog is here:
> https://github.com/HSF/phoenix/tree/master/packages/phoenix-ng/projects/phoenix-ui-components/src/components/ui-menu/collections-info/collections-info-overlay
> 
> So the idea would be that for each column, we add an arrow to allow sorting the list by that column. The default should be the number (i.e. the current behaviour).

I cannot find the exact tables there

**Comment by 9inpachi** [comment]
The table is at [collections-info-overlay.component.html#L31](https://github.com/HSF/phoenix/blob/master/packages/phoenix-ng/projects/phoenix-ui-components/src/components/ui-menu/collections-info/collections-info-overlay/collections-info-overlay.component.html#L31) and the columns are generated at run time from data at [collections-info-overlay.component.html#L41](https://github.com/HSF/phoenix/blob/master/packages/phoenix-ng/projects/phoenix-ui-components/src/components/ui-menu/collections-info/collections-info-overlay/collections-info-overlay.component.html#L41).

**Comment by Ayushsunny** [comment]
Hello @EdwardMoyse , @9inpachi I am trying to add sorting on that component file by 
this as a html component
```html
<th *ngFor="let column of options.columns" (click)="sortHeaderClick(column.value)"
  [ngClass]="{ 'sorting': isSorting(column.value), 'sorting_asc': isSortAsc(column.value), 'sorting_desc': isSortDesc(column.value) }">
  <span [innerHTML]="column.name"></span>
</th>
```

and this in the typescript component 

```typescript
private sort(array: Array<any>, fieldName: string, direction: string, isNumeric: bool)
{
  var sortFunc = function (field, rev, primer) {
      // Return the required a,b function
      return function (a, b) {
          // Reset a, b to the field
          a = primer(pathValue(a, field)), b = primer(pathValue(b, field));
          // Do actual sorting, reverse as needed
          return ((a < b) ? -1 : ((a > b) ? 1 : 0)) * (rev ? -1 : 1);
      }
  };
 
  // Have to handle deep paths
  var pathValue = function (obj, path) {
      for (var i = 0, path = path.split('.'), len = path.length; i < len; i++) {
          obj = obj[path[i]];
      };
      return obj;
  };
 
  var primer = isNumeric ?
      function (a) {
          var retValue = parseFloat(String(a).replace(/[^0-9.-]+/g, ''));
          return isNaN(retValue) ? 0.0 : retValue;
      } :
      function (a) { return String(a).toUpperCase(); };
 
  this._isSorting = true;
  this._start = new Date().getTime();
  array.sort(sortFunc(fieldName, direction === 'desc', primer));
  this._end = new Date().getTime();
  var time = this._end - this._start;
  console.log('Sort time: ' + time);
  this._isSorting = false;
}
```

**Comment by Ayushsunny** [comment]
yarn compiling perfectly without any problem but code was not implemented as what we needed.
Actually I am new to Angular.
could you guide with me with that please Sir.
as I really wanted to be a long time contributor in this organization

**Comment by 9inpachi** [comment]
Hi @Ayushsunny, this code will not work since it has not been adapted to work with the `collections-info-overlay` component (aka it's copy pasted). I don't see anything in the Angular template file that makes a call to the actual TypeScript code for sorting the elements.

Please try to go through the basic Angular tutorials and understand how Angular works - [Getting started with Angular](https://angular.io/start).

**Comment by Ayushsunny** [comment]
> Hi @Ayushsunny, this code will not work since it has not been adapted to work with the `collections-info-overlay` component (aka it's copy pasted). I don't see anything in the Angular template file that makes a call to the actual TypeScript code for sorting the elements.
> 
> Please try to go through the basic Angular tutorials and understand how Angular works — [Getting started with Angular](https://angular.io/start).
Hello, @9inpachi I appreciate that sir for the help,
Actually, I have done that Getting started tutorial earlier as I was exploring this new lib but I couldn't found the sorting method there.
By the way, I got one Angular header sorting tutorial [here](https://material.angular.io/components/sort/overview) ---is this the thing which we're going to add there? 
If no then please tell me where to find the sorting tutorial for this repo or If yes then may I add this with custom collection data?

**Comment by 9inpachi** [comment]
You are not going to find the "solution" somewhere, you have to code it. The code snippet you shared above clearly shows that you haven't gone through the basics the right way because it's completely wrong and taken out of context from somewhere else. I will suggest to go through the Angular guidelines again and become familiar with Angular by building something of your own and later come back to understand how sorting could be done for Phoenix. Just copy pasting isn't going to get anything done.

**Comment by Yash621** [comment]
Hello Sir @EdwardMoyse, can you guide me on this? I would like to work on this issue

**Comment by gauravsharma21** [comment]
Hi, I am new to open source contribution and worked upon this issue. Please review my pull request https://github.com/HSF/phoenix/pull/377 
Any suggestions on improvement upon my PR would be great.

**Comment by 9inpachi** [comment]
Closed with #377
